sync/src/synchronizer/block_process.rs

```diff
@@ -32,14 +34,87 @@ impl<'a> BlockProcess<'a> {
         let shared = self.synchronizer.shared();
         let state = shared.state();
 
-        if state.new_block_received(&block) {
-            if let Err(err) = self.synchronizer.process_new_block(block.clone()) {
+        if !shared.active_chain().is_initial_block_download() {
+            // determine should we drop the queue and consumer handle
+
+            let queue_is_empty = self
+                .synchronizer
+                .block_queue
+                .read()
+                .map(|queue_op| {
+                    queue_op
+                        .as_ref()
+                        .map(|queue| queue.is_empty())
+                        .unwrap_or(false)
+                })
+                .unwrap_or(false);
+
+            if queue_is_empty {
+                // drop block_queue and consumer handle
+                let _ = self
```

Comment:
I suggest using `expect` directly here, like `read().expect("block_queue lock poisoned on a panic")`
So when this happens unexpectedly, we will be able to know a panic occurs while a write operation is in progress.
let_underscore is also bad, although there is some let_underscore existing now.